apps/meteor/app/apps/server/bridges/rooms.ts

```diff
@@ -245,6 +245,60 @@ export class AppRoomBridge extends RoomBridge {
 		return users.map((user: ICoreUser) => userConverter.convertToApp(user));
 	}
 
+	protected async getUnreadByUser(roomId: string, userId: string, options: GetMessagesOptions, appId: string): Promise<Array<IMessageRaw>> {
+		this.orch.debugLog(`The App ${appId} is getting the unread messages for the user: "${userId}" in the room: "${roomId}"`);
+
+		const messageConverter = this.orch.getConverters()?.get('messages');
+		if (!messageConverter) {
+			throw new Error('Message converter not found');
+		}
+
+		const room = await Rooms.findOneById(roomId, { projection: { _id: 1 } });
+
+		if (!room) {
+			throw new Error('Room not found');
+		}
+
+		const subscription = await Subscriptions.findOneByRoomIdAndUserId(roomId, userId, { projection: { ls: 1 } });
+
+		const lastSeen = subscription?.ls;
+		if (!lastSeen) {
+			return [];
+		}
+
+		const sort: Sort = options.sort?.createdAt ? { ts: options.sort.createdAt } : { ts: 1 };
+
+		const cursor = Messages.findVisibleByRoomIdBetweenTimestampsNotContainingTypes(roomId, lastSeen, new Date(), [], {
+			...options,
+			sort,
+		});
+
+		const messages = await cursor.toArray();
+		return Promise.all(messages.map((msg) => messageConverter.convertMessageRaw(msg)));
```

Comment:
map could happen on the cursor iirc. 

return cursor.map(yourFunc).toArray();

Reply:
Thank you for your suggestion. I understand that mapping directly on the cursor can be more concise. However, I prefer the current approach for the following reasons:

1. Clarity
2. Readability

Please elaborate on the benefits you see in mapping directly on the cursor. Is it something standard we started to follow? I haven't come across any documentation mentioning this, so if it's more of a personal preference, I'd prefer to keep it as it is.

Reply:
we do this everywhere last I remember. cursor mapping means it's truely an iterator-like. not doubling the memory allocation.

https://github.com/RocketChat/Rocket.Chat/blob/d0490d3d84193b3a9ceec899673757209c14de3b/apps/meteor/app/apps/server/bridges/rooms.ts#L133-L135

you're also one of the folks following the same pattern in the past

Reply:
Oh, my bad! But to be honest, I am not sure we did this everywhere; IMHO, it doesn't matter much.